### PR TITLE
Oc 10523

### PR DESCRIPTION
**Pull request by slclinica**

### CODE REVIEW

File: src/main/java/core/org/akaza/openclinica/dao/hibernate/JobDetailDao.java [hunk 1]
```diff
@@ -44,5 +44,11 @@ public JobDetail findByUuid(String uuid) {
         return (JobDetail) q.uniqueResult();
     }
 
-
+    public List<JobDetail> findStudyIdWithStatus(int studyId, Enum status) {
```
Comment (svadla-oc) [inline src/main/java/core/org/akaza/openclinica/dao/hibernate/JobDetailDao.java]: can you rename this to findByStudyIdAndStatus?

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java [hunk 2]
```diff
@@ -29,85 +41,110 @@
...
+
+    @RequestMapping(value = "/studies/{studyEnvUuid}/publishJobs", method = RequestMethod.POST)
+    public ResponseEntity<String> importOdmToOC(@PathVariable("studyEnvUuid") String studyEnvUuid, @RequestBody PublishDTO publishDTO, HttpServletRequest request)
+            throws Exception {
 
         String accessToken = (String) request.getSession().getAttribute("accessToken");
+        Study study = studyDao.findByStudyEnvUuid(studyEnvUuid);
+        String studyOid = study.getOc_oid();
+        utilService.setSchemaFromStudyOid(studyOid);
+        study = studyDao.findByStudyEnvUuid(studyEnvUuid);
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: can you add some comment explaining why you are fetching the study again?

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java [hunk 3]
```diff
@@ -29,85 +41,110 @@
...
             ODM odm = publishDTO.getOdm();
-            List <Page> pages = publishDTO.getPages();
-            Map<String, Object> map = null;
+            List<Page> pages = publishDTO.getPages();
             try {
                 CoreResources.tenantSchema.set("public");
-                map = (Map<String, Object>) odmImportService.importOdm(odm, pages, publishDTO.getBoardId(), accessToken);
+                map = odmImportService.importOdm(odm, pages, publishDTO.getBoardId(), accessToken);
             } catch (Exception e) {
+                utilService.setSchemaFromStudyOid(studyOid);
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: do we need to switch the schema here again?

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java [hunk 4]
```diff
@@ -29,85 +41,110 @@
...
+                userService.persistJobFailed(jobDetail, fileName);
                 throw new CompletionException(e);
             }
-            Study study = (Study) map.get("study");
-            Study publicStudy = studyDao.findPublicStudy(study.getOc_oid());
+            Study publicStudy = studyDao.findPublicStudy(studyOid);
             odmImportService.updatePublicStudyPublishedFlag(publicStudy);
             odmImportService.setPublishedVersionsInFM(map, accessToken);
-            return new ResponseEntity<Object>(null, HttpStatus.OK);
+            utilService.setSchemaFromStudyOid(studyOid);
```
Comment (svadla-oc) [inline src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: do we need to switch the schema here again?

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java (hunk 2 above)
Comment (slclinica) [reply src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: yup!

File: src/main/java/core/org/akaza/openclinica/dao/hibernate/JobDetailDao.java (hunk 1 above)
Comment (slclinica) [reply src/main/java/core/org/akaza/openclinica/dao/hibernate/JobDetailDao.java]: yup!

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java (hunk 3 above)
Comment (slclinica) [reply src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: nope.

File: src/main/java/org/akaza/openclinica/controller/OdmImportController.java (hunk 4 above)
Comment (slclinica) [reply src/main/java/org/akaza/openclinica/controller/OdmImportController.java]: nope.

**Comment by svadla-oc** [comment]
👍

**Pull request merged**